src/main/java/org/sagebionetworks/web/client/widget/entity/EntityPageTop.java

```diff
@@ -467,6 +504,7 @@ private void fileChanged(EntityBundle bundle, Long version) {
 
 	public void showSelectedTabs() {
 		visibleTabCount = 0;
+		countTabContentChecked = 0;
```

Comment:
This is where we start to determine what tabs are visible.  count is incremented by each callback.  On the last callback (when countTabContentChecked == EntityArea.values().length), we check to see if the currently selected area tab is visible.  If not, change the area to a visible area and reconfigure.

Reply:
Got it...